### DISCUSSION TITLE
Linux: Unable to save existing "read-only" file (it isn't) on NAS

### DISCUSSION THREAD

**Discussion by mihawk90** [SubtitleEdit/subtitleedit#8627] [Q&A]
So this is a strange one that's been bothering me for a while. I just updated to 4.0.7 as well, issue persists.

I've been OCRing a lot of my TV show PGS subs lately and saving them directly to my NAS (mounted via NFS), which both works great. However, occasionally mistakes slip through that I only notice when actually watching the show with subtitles. After fixing the mistakes within SE however, I am unable to save the file:

> Cannot save /mnt/tank/media/tvshows/The Office [tmdbid-2316]/Season 04/\<file\>.srt
> 
> File is read-only!

However, I know for a fact it is not. I can open the exact same file in Kwrite (or even just `vim`), edit, and save it just fine.

I'm not sure if this is an SE specific issue, or if Mono is to blame.
As noted above the volume is mounted via NFS. All users accessing the share are mapped to a user on the NAS that has read and write access, but it does have a different user ID. Is it possible that Mono doesn't check the actual file permissions, but only compares the user ID instead? Other then that I wouldn't know what could cause this...

This is what permissions look like:

```
 id # local user
uid=1000(tarulia) gid=1000(tarulia) groups=1000(tarulia),10(wheel),135(mock) context=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023
 pwd
/mnt/tank/media/tvshows/The Office [tmdbid-2316]/Season 04
 getfacl <file>.srt
# file: <file>.srt
# owner: 3000
# group: 3001
user::rw-
group::rwx                      #effective:rw-
group:3001:rwx                  #effective:rw-
mask::rw-
other::r--
```

Anybody have an idea? :/

As a workaround I currently save a second file and then delete the first but.. that's not very comfortable :'D

Any help would be appreciated :)

**Comment by pcb-man** [comment]
I have the same issue. If the files are copied to a "local" folder, say to "documents", it has no problem writing. Based on that, I think it's getting its permissions from the wrong place.  Still have not found a solution. I notice that in the "file open" dialog, the NSF share doesn't show up. In order to make it show up, I have to make a link to it in a place that DOES show, like the desktop. I really want to fix this issue, because SubtitleEdit is by far my favorite utility for subtitles.
UPDATE: Boy, am I dumb! The file I was editing IS read only for everybody but root, but NFS does not allow root access from the client on purpose. Instead of editing and saving the file back to itself, I tried "save as", and it did that just fine. That's actually a good safety measure as far as I'm concerned. Just a change in workflow.  The "save as" file can be edited and saved onto itself just fine. I could go change the rw permissions on the file, but I think I'll leave as-is for safety.

**Reply by mihawk90** [reply]
> I notice that in the "file open" dialog, the NSF share doesn't show up.

Not sure what you mean here? You mean there's no "drive" under "My Computer"? Network mounts aren't drives, so them not showing up there is correct. My NFS mount does show up under "My Network" though:
![image](https://github.com/user-attachments/assets/f64944dc-a5e7-4a9a-9074-6e2354e1e130)

> I tried "save as", and it did that just fine

I'm not sure why that works for you. From a permissions standpoint that doesn't really make sense. Whether you actively select the file or just continue using the one already selected makes no difference in permissions. At the end of the day It's writing to the same file descriptor, which also has the same permissions.
Either way, that doesn't work for me either, even though the file *is* read-write as seen above. Unless for some reason .NET is using the `other` permission somehow.
Normally .NET shouldn't even be checking the permissions though. It just tries opening the file as read-write and either gets a read-write file descriptor back from the filesystem or the access is denied.

I also just checked to make sure and mono is definitely running as my user:
```
 ps aux | grep SubtitleEdit
tarulia    18667  1.3  3.1 2965872 507884 ?      Ssl  03:20   0:10 /usr/bin/mono /home/tarulia/Applications/SubtitleEdit/SubtitleEdit.exe
```

And as noted that user can write to the file using Vim or KWrite just fine.
